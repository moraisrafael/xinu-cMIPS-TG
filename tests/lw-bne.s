	.include "cMIPS.s"
	.text
	.align 2
	.globl _start
	.ent _start
_start: nop
	la   $15,x_DATA_BASE_ADDR
        la   $16,x_IO_BASE_ADDR
	li   $3,1
	li   $4,2
	li   $22,40
incr:	mult $3,$4
	mflo $5
	addiu $15,$15,4
	sw   $5, 0($15) # ram[0]=2,4,6,8,...,x20,x22,x24,x26,x28
	addi $3,$3,1
	sw   $5, 0($16)
	lw   $7,0($15)
	bne  $22,$7,incr 
	nop
	nop
	nop
	nop
	wait
	.end _start

