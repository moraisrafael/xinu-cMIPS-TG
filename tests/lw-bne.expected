00000002
00000004
00000006
00000008
0000000a
0000000c
0000000e
00000010
00000012
00000014
00000016
00000018
0000001a
0000001c
0000001e
00000020
00000022
00000024
00000026
00000028
